### PR TITLE
Add conda builds to our github actions workflow

### PR DESCRIPTION
**Pull request by benfred**
This adds a conda build step to build our conda package to the github actions workflow.
This also uploads the package to anaconda during release.

### CODE REVIEW

File: .github/workflows/cpu-ci.yml [hunk 1]
```diff
@@ -14,11 +14,13 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        python-version: [3.8]
+        python-version: [3.7, 3.8, 3.9]
```
Comment (karlhigley) [inline .github/workflows/cpu-ci.yml:17]: RAPIDS has dropped support for 3.7, so we have too. We're starting to make active use of Python 3.8 features like `typing.Protocol`, so might want to omit 3.7 here.

File: .github/workflows/cpu-ci.yml [hunk 2]
```diff
@@ -14,11 +14,13 @@ jobs:
...
         os: [ubuntu-latest]
 
     steps:
     - uses: actions/checkout@v2
+      with:
+       fetch-depth: 0
```
Comment (karlhigley) [inline .github/workflows/cpu-ci.yml:23]: 👍🏻

**Review by karlhigley — APPROVED** [review]

File: .github/workflows/cpu-ci.yml (hunk 1 above)
Comment (benfred) [inline .github/workflows/cpu-ci.yml:17]:
```suggestion
        python-version: [3.8, 3.9]
```

Comment (benfred) [reply .github/workflows/cpu-ci.yml]: done

**Comment by benfred** [comment]
rerun tests

**Pull request merged**